src/NzbDrone.Core/Indexers/HttpIndexerBase.cs

```diff
@@ -493,19 +493,27 @@ protected virtual async Task<ValidationFailure> TestConnection()
 
                 return new ValidationFailure(string.Empty, "Unable to connect to indexer. " + ex.Message);
             }
-            catch (HttpException ex)
+            catch (WebException ex)
             {
-                if (ex.Response.StatusCode == HttpStatusCode.BadRequest &&
-                    ex.Response.Content.Contains("not support the requested query"))
+                if (ex.Status == WebExceptionStatus.NameResolutionFailure || ex.Status == WebExceptionStatus.ConnectFailure)
                 {
-                    _logger.Warn(ex, "Indexer does not support the query");
-                    return new ValidationFailure(string.Empty, "Indexer does not support the current query. Check if the categories and or searching for movies are supported. Check the log for more details.");
+                    _logger.Warn("{0} server could not be reached. {1}", this, ex.Message);
+                    return new ValidationFailure(string.Empty, string.Format("{0} server could not be reached. {1}", this, ex.Message));
                 }
-                else
+            }
+            catch (HttpException ex)
+            {
+                if (ex.Message.Contains("502") || ex.Message.Contains("503") ||
+                    ex.Message.Contains("timed out"))
                 {
-                    _logger.Warn(ex, "Unable to connect to indexer");
+                    _logger.Warn("{0} server is currently unavailable. {1} {2}", this, ex.Request.Url, ex.Message);
+                    return new ValidationFailure(string.Empty, string.Format("{0} server is currently unavailable. {1} {2}", this, ex.Request.Url, ex.Message));
+                }
 
-                    return new ValidationFailure(string.Empty, "Unable to connect to indexer, check the log for more details");
+                if (ex.Response.StatusCode == HttpStatusCode.BadRequest && ex.Response.Content.Contains("not support the requested query"))
+                {
+                    _logger.Warn(ex, "Indexer does not support the query");
+                    return new ValidationFailure(string.Empty, "Indexer does not support the current query. Check the log for more details.");
                 }
```

Comment:
Still need the else on this right? or else any other HttpException won't return as a ValidationFailure